synapse/storage/schema/main/delta/78/03_rooms_to_delete.sql

```diff
@@ -13,14 +13,15 @@
  * limitations under the License.
  */
 
--- cf upsert_room_to_purge docstring for the meaning of the fields.
-CREATE TABLE IF NOT EXISTS rooms_to_purge(
+-- cf upsert_room_to_delete docstring for the meaning of the fields.
+CREATE TABLE IF NOT EXISTS rooms_to_delete(
```

Comment:
What's the context behind switching from `rooms_to_purge` -> `rooms_to_delete` everywhere?

We still have the concept of purging rooms in the code base

Reply:
`rooms_to_delete` seems like a misnomer now since this encompasses 3 difference cases: history that needs to be purged, rooms that need to be purged, and rooms that need to be shutdown and purged.

It's kinda turning into a generic task queue to schedule and retry things (state is stored in the database). Do we have any pattern to cover this already somewhere else?

Reply:
We have the `background_updates` used for the DB upgrades. I think the rest of the background jobs are not stored in DB and not resumable. 

You are right that in the end this table is generic enough to have a generic scheduling task framework around it.

I am happy to do that, I however don't really want to mix that up with the `background_updates` one.